crates/repository/src/lib.rs

```diff
@@ -421,10 +422,10 @@ impl Repository {
 
 async fn download_with_fallback(
     output_path: impl AsRef<OsStr>,
-    urls: Vec<String>,
+    urls: impl IntoIterator<Item = impl AsRef<OsStr> + Display>,
     connect_timeout: Duration,
 ) -> Result<()> {
-    for (i, url) in urls.iter().enumerate() {
+    for (i, url) in urls.into_iter().enumerate() {
         if i > 0 {
             println!("Falling back to downloading from {url}");
```

Comment:
This is a weird constraint. `OsStr` explicitly is not `Display`. I propose the following:
```rust
    urls: impl IntoIterator<Item = impl AsRef<OsStr>>,
    connect_timeout: Duration,
) -> Result<()> {
    for (i, url) in urls.into_iter().enumerate() {
        let url = url.as_ref();
        if i > 0 {
            println!("Falling back to downloading from {url:?}");
        }
```

Reply:
Wouldn't this require the Debug trait (`+ Debug`)? Why is this preferred in this case?

Reply:
No, you missed the `let url = url.as_ref();`. `OsStr` implements `Debug`.

Reply:
Indeed, I totally missed that, fixed now.